docs/admin/deployment.md

```diff
@@ -715,6 +720,17 @@ For class == "nvme", the following parameters should be populated:
 
 - `bdev_list` should be populated with NVMe PCI addresses.
 
+The default way for DAOS to access NVMe storage is through SPDK via the VFIO user-space driver.
+To use an alternative driver with SPDK, set `disable-vfio: true` in the global section of the
+server config file to fallback to using UIO user-space driver with SPDK instead.
+If IOMMU and VFIO are not enabled in the BIOS, the alternative driver will be used by default.
+
+If VMD is enabled on a host, its usage will be enabled by default meaning that the `bdev_list`
```

Comment:
my experience is that the daos_server will try to use VMD even if it is not enabled on the platform. we need a warning for people updating to 2.2 from 2.0 that the old `enable_vmd:true/false` is no longer honoured, and instead the `disable_vmd:false/true` needs to be specified. otherwise, the server does not come up after an update from 2.0 to 2.2.

Reply:
Can you elaborate on "daos_server will try to use VMD even if it is not enabled on the platform"? on master and 2.2 branches, VMD should only be being activated if platform-enabled. On 2.0 it should only be activated if platform-enabled and enable_vmd set true in config file. if this isn't true then can you please raise a ticket?

Reply:
I will add the warning to the admin guide, is that okay?

Reply:
Hm, I wonder if we should fix this behavior in 2.2.  Seems like we should allow enable_vmd as a legacy config option like we do with the old pre-tiering storage options. It's ugly from a code perspective but I'd expect a minor version upgrade to be backwards-compatible with config files from the same major version, at least.

Reply:
I don't want to do that as part of this PR as it is a doc-change.

If we do want to add backwards capability then how would that work? you would have both enable and disable vmd flags and the default option would always be to "not enable" because the unset value of the enabled flag would be false. @mjmac @Michael-Hennecke do we want to pursue this as a separate issue?

Reply:
https://github.com/daos-stack/daos/pull/9415